core/types/transaction.go

```diff
@@ -442,13 +442,15 @@ func (t *TransactionsFixedOrder) Peek() Transaction {
 // Shift replaces the current best head with the next one from the same account.
 func (t *TransactionsFixedOrder) Shift() {
 	t.Transactions = t.Transactions[1:]
+	t.Transactions[0] = nil // avoid memory leak
 }
 
 // Pop removes the best transaction, *not* replacing it with the next one from
 // the same account. This should be used when a transaction cannot be executed
 // and hence all subsequent ones should be discarded from the same account.
 func (t *TransactionsFixedOrder) Pop() {
 	t.Transactions = t.Transactions[1:]
+	t.Transactions[0] = nil // avoid memory leak
```

Comment:
need tmp variable, this assignment to wrong place

Reply:
like this?
https://github.com/ledgerwatch/erigon/pull/6204/commits/31ac2c6e7d06782b34d0a9f0a4de2f4bce5f39dc

Reply:
btw, are `Shift()` and `Pop()` supposed to be the same?

Reply:
@estensen this class code is from Geth, I advise you to see original implementation there.